polkadot/xcm/xcm-builder/src/barriers.rs

```diff
@@ -82,10 +82,10 @@ impl<T: Contains<Location>> ShouldExecute for AllowTopLevelPaidExecutionFrom<T>
 		instructions[..end]
 			.matcher()
 			.match_next_inst(|inst| match inst {
-				WithdrawAsset(ref assets) |
-				ReceiveTeleportedAsset(ref assets) |
-				ReserveAssetDeposited(ref assets) |
-				ClaimAsset { ref assets, .. } =>
+				Instruction::WithdrawAsset(WithdrawAsset(ref assets)) |
```

Comment:
@acatangiu this is a big annoying breaking change. I want to confirm if this is fine.
Otherwise we can create a big tedious enum adapter to reduce the breaking change.

Reply:
AFAICT the behavior does not change and the public APIs / usage also doesn't change, so this is breaking for who?

afaict it's not breaking for downstream consumers of polkadot-sdk - the major sdk release containing will also contain all of XCMv6, all components internally adapted to XCMv6, including these barriers.

If it's a problem that it is breaking internally as in it introduces too many internal changes to adapt to the new structs, then I am not concerned. I am ok with the verbose changes as well as extra adapter to reduce diffs.

Reply:
the Instruction enum is part of the public Rust API
I changed it to be compatible to minimize amount of code change needed to support v6